### PR TITLE
Reviews

### PR DESCRIPTION
**Pull request by MahaBKT**

### CODE REVIEW

File: src/com/marketplace/service/review/ReviewRepresentation.java [hunk 1]
```diff
@@ -1,5 +1,41 @@
...
+
+
 
+@XmlRootElement(name = "Review")
+@XmlAccessorType(XmlAccessType.FIELD)
+@XmlType(name = "")
+
+public class ReviewRepresentation {
+	
+	private int reviewID;
```
Comment (lladewig) [inline src/com/marketplace/service/review/ReviewRepresentation.java]: Missing some fields from the `Review` object. Checkout `Review.java` to see what's missing

File: src/com/marketplace/service/review/ReviewRepresentation.java [hunk 2]
```diff
@@ -1,5 +1,41 @@
...
+		return reviewScore;
+	}
+	public void setReviewScore(int reviewScore) {
+		this.reviewScore = reviewScore;
+	}
+	public String getReviewBody() {
+		return reviewBody;
+	}
+	public void setReviewBody(String reviewBody) {
+		this.reviewody = reviewBody;
```
Comment (lladewig) [inline src/com/marketplace/service/review/ReviewRepresentation.java]: `this.reviewody` -> `this.reviewBody` - just a small typo!

File: src/com/marketplace/service/review/ReviewActivity.java [hunk 3]
```diff
@@ -1,5 +1,83 @@
...
+		rRes.setcustomer(review.getcustomer());
+		rRes.setproduct(review.getproduct());
+		rRes.setReviewScore(review.getReviewScore());
+		rRes.setReviewBody(review.getReviewBody());	
+		return rRes;
+	}
+	
+	public ReviewRepresentation updateReviewScore(ReviewRequest rReq) {
+		ReviewLogic rLogic = new ReviewLogic();
+		Review review = rLogic.updateReviewScore(rReq.getReviewScore(), rReq.getreviewID());
```
Comment (lladewig) [inline src/com/marketplace/service/review/ReviewActivity.java]: `rReq.getreviewID()` -> `rReq.getReviewID()`

File: src/com/marketplace/service/review/ReviewResource.java [hunk 4]
```diff
@@ -1,5 +1,43 @@
...
+		}
+		System.out.println("GET METHOD Request from Client for Get All Reviews with offset " + offset + " and limit " + limit);
+		ReviewActivity rActivity = new ReviewActivity();
+		return rActivity.getAllReviews(offset, limit);
+	}
+	
+	@POST
+	@Produces({"application/xml" , "application/json"})
+	@Path("/reviews")
+	public ReviewsRepresentation rCustomer(ReviewRequest rReq) {
```
Comment (lladewig) [inline src/com/marketplace/service/review/ReviewResource.java]: I'm not sure what this method does.

File: src/com/marketplace/service/review/ReviewResource.java [hunk 5]
```diff
@@ -1,5 +1,43 @@
...
+		System.out.println("POST METHOD Request from Client to Add Reviews");
+		ReviewActivity rActivity = new ReviewActivity();
+		return rActivity.rCustomer(rReq);
+	}
+	
+	}
 }
```
Comment (lladewig) [inline src/com/marketplace/service/review/ReviewResource.java]: Missing `addReview`, `deleteReview`, and `updateReview`

**Comment by MahaBKT** [comment]
fixed

**Pull request closed without merging**